hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -27,38 +31,53 @@ export default function HackathonSelectDropdown(setHackathonID: Props) {
     )
   }
 
-  const hackathonMap = hackathonList.map((hackathon, index) => ({
+  const hackathonDateCurrentAndFuture = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) > today
+  })
+
+  const hackathonDatePast = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) < today
+  })
+
+  const hackathonMap = (
+    context === 'archive'
+      ? hackathonDatePast
+      : context === 'idea-portal'
+      ? hackathonDateCurrentAndFuture
+      : hackathonList
+  ).map((hackathon, index) => ({
     value: hackathon.id,
     label:
       hackathon.title +
       ' ' +
-      new Date(hackathon.startDate).toLocaleDateString() +
+      new Date(hackathon.startDate).toLocaleString(undefined, {
+        year: '2-digit',
+        month: '2-digit',
+        day: '2-digit',
+      }) +
       '-' +
-      new Date(hackathon.endDate).toLocaleDateString(),
+      new Date(hackathon.endDate).toLocaleString(undefined, {
+        year: '2-digit',
+        month: '2-digit',
+        day: '2-digit',
+      }),
```

Comment:
Let's extract this so it's more readable and the result type is clear
```
const hackathonMap = getHackatonsForContext()
    .map(hackathon => mapHackatonToSelectItem(hackathon))
```
example impl:
```
function mapHackatonToSelectItem(hackathon: HackathonPreview): SelectItem {
    return {
      value: hackathon.id,
      label: hackathon.title +
        ' ' +
        new Date(hackathon.startDate).toLocaleDateString() +
        '-' +
        new Date(hackathon.endDate).toLocaleDateString()
    }
  }
```
Note you can use `toLocaleDateString()` to return just the date